src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlBindingPathTransformer.cs

```diff
@@ -99,18 +99,18 @@ IEnumerable<XamlPropertyAssignmentNode> getResourceValues(IXamlAstNode node)
                     }
                 }
 
-                if (startType == null)
+                Func<IXamlType> startTypeResolver = startType is not null ? () => startType : () =>
```

Comment:
Wouldn't it be better to just use Lazy<T>?

Reply:
`Lazy<T>` has all this extra internal machinery to support thread safety that we don't need as we use this once. I think directly using a `Func<T>` is a little cleaner.

Reply:
Ah, it's only being called once. It's ok then